### DISCUSSION TITLE
Welcome to Tween.js Discussions!

### DISCUSSION THREAD

**Discussion by trusktr** [tweenjs/tween.js#595] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by AsDeadAsADodo** [comment]
So i have this scenario : two aeroplanes fly side by side and in the same altitude.let's call them aeroplaneA and aeroplaneB.There are two phase of their movement : the first one is that aeroplaneA and aeroplaneB maintain the movement,the other one is that aeroplaneB change its altitude fly down 10 unit while maintain the Horizontal speed.

```javascript
 let groupA = new Group();
 let groupB = new Group();
 const speed = 300;
 const fullTime = 10; 
 let aeroplaneA_Tween = new TWEEN(aeroplaneA.position,groupA);
 let aeroplaneB_Tween_phase_one = new TWEEN(aeroplaneB.position,groupB);
 let aeroplaneB_Tween_phase_two = new  TWEEN(aeroplaneB.position,groupB);
 aeroplaneA_Tween.to({
 	z:aeroplaneA.position.z+speed*fullTime
 	},fullTime);
 aeroplaneB_Tween_phase_one.to({
 	z:aeroplaneB.position.z+speed*fullTime/2
	},fullTime/2*1000);
 aeroplaneB_Tween_phase_two.to({
 	y : aeroplaneB.position.y-10 ,
 	z : aeroplaneB.position.z+speed*fullTime/2
	},fullTime/2*1000);
	aeroplaneB_Tween_phase_one.chain(aeroplaneB_Tween_phase_two);
	
	aeroplaneA_Tween.start();
	aeroplaneB_Tween_phase_one.start();
	
	......
	
	function animate(){
		groupA.update();
		groupB.update();
		...... //render canvas
		requestAnimationFrame(animate);
	}
```

But the thing is , at the very beginning of the phase two , I can see clearly aeroplaneB is fly behind aeroplaneA , they suppose to fly side by side .
I guess its because aeroplaneA is continue its movement while aeroplaneB phase two code is not executed fully.
AeroplaneA and aeroplaneB are simply not synchronized .
I dont know for sure . but if my deduction is right , how do i synchronized them ?

**Reply by trusktr** [reply]
Hello @AsDeadAsADodo. Thanks for realizing to make a new thread. I replied there.
